src/modules/person/persistence/person.repository.integration-spec.ts

```diff
@@ -860,6 +861,125 @@ describe('PersonRepository Integration', () => {
                 await expect(sut.update(person)).rejects.toBeDefined();
             });
         });
+        describe('when referrer is defined', () => {
+            it('should use the existing referrer if the person has not been renamed', async () => {
+                const existingPerson: Person<true> = await savePerson();
+                kcUserServiceMock.setPassword.mockResolvedValueOnce({
+                    ok: true,
+                    value: 'mockedPassword',
+                });
+                const result: Person<true> | DomainError = await sut.update(existingPerson);
+
+                expect(result).toBeInstanceOf(Person);
+                if (result instanceof Person) {
+                    expect(result.referrer).toEqual(existingPerson.referrer);
+                }
+            });
+        });
+        describe('when referrer is undefined', () => {
+            afterAll(() => {
+                jest.restoreAllMocks();
+            });
```

Comment:
Should this be `afterAll` or `afterEach`? We usually restore the mocks in `beforeEach`